core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1863,26 +1870,28 @@ private[spark] class DAGScheduler(
           // TODO: mark the executor as failed only if there were lots of fetch failures on it
           if (bmAddress != null) {
             val externalShuffleServiceEnabled = env.blockManager.externalShuffleServiceEnabled
-            val isHostDecommissioned = taskScheduler
-              .getExecutorDecommissionState(bmAddress.executorId)
-              .exists(_.workerHost.isDefined)
+            val isHostDecommissioned = execIdOpt.exists { execId =>
+              taskScheduler
+                .getExecutorDecommissionState(execId)
+                .exists(_.workerHost.isDefined)
+            }
 
-            // Shuffle output of all executors on host `bmAddress.host` may be lost if:
+            // Shuffle output of all executors on host `bmId.host` may be lost if:
```

Comment:
Shall we revert this? In this context, we have `bmAddress` instead of `bmId`, don't we?

Reply:
Ah..good catch! The original change has been reverted, we should revert the comment as well.